drivers/bluetooth/hci/rpmsg.c

```diff
@@ -317,9 +325,55 @@ static int bt_rpmsg_open(void)
 	return 0;
 }
 
+static int bt_rpmsg_close(void)
+{
+	int err;
+	struct net_buf *buf = NULL;
+	struct net_buf *rsp = NULL;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_RESET, 0);
+	if (!buf) {
+		LOG_ERR("Allocating command buffer failed!");
+		return -ENOBUFS;
+	}
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, &rsp);
```

Comment:
There's a buffer leak here, since you're actually not doing anything with the allocated `buf` (not even calling `net_buf_unref()` on it). You actually don't need the `buf` and `rsp` buffers at all if you just intend to send a command without additional parameter and don't care about the contents of the command complete event either. You can simply do:
```c
err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, NULL);
```

Reply:
Good point, reworked it as you said.